File: code/online_challenges/src/codechef/DYNAMO/Dynamo.java
import java.util.Scanner;

class Dynamo {
    public static void main (String[] args)
    {
        Scanner s = new Scanner(System.in);
        int t = s.nextInt();
        while (t-- > 0){
            int n = s.nextInt();
            long a = s.nextLong();

            long S = (long)(2 * Math.pow(10, n)) + a;
            System.out.println(S);

            long b = s.nextLong();

            long c = (long)(Math.pow(10, n)) - b;
            System.out.println(c);

            long d = s.nextLong();

            long e = S - a - b - c - d;
            System.out.println(e);

            int ans = s.nextInt();
            if (ans == -1) break;

            System.out.println();
            System.out.flush();
        }
    }
}

/*
EXAMPLE

You             Grader
                1
                2
                10
50
                10
12
                10
8
                1

*/

File: code/online_challenges/src/codechef/DYNAMO/Dynamo.cpp
#include <bits/stdc++.h>
#include<math.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    while ( t -- ) {
        
        int n, x;
        long long int a, b, c, d, e, s;
        cin >> n;
        long long int ten = pow(10,n);
        cin >> a;
        cout << (2 * ten + a) << "\n";
        cout.flush();
        cin >> b;
        cout << ( ten - b ) << "\n";
        cout.flush();
        cin >> d;
        cout << ( ten - d ) << "\n";
        cout.flush();
        cin >> x;

        if(x == -1) return 0;
    }
    return 0;
}

